docs/core/deploying/native-aot/diagnostics.md

```diff
@@ -0,0 +1,70 @@
+---
+title: AOT diagnostics
+description: Learn about diagnostics in native AOT applications
+author: lakshanf
+ms.author: lakshanf
+ms.date: 08/07/2023
+---
+# Diagnostics in native AOT applications
+
+The long-term goal for native AOT application diagnostics is to provide the rich diagnostic experience that developers expect out of a .NET application. The .NET diagnostic experience was built over multiple releases, and a critical part of that journey was customer feedback. Native AOT diagnostics will follow a similar path where the right diagnostic experience is built from prioritized customer feedback in multiple releases.
+
+## Native AOT application features
+
+Native AOT applications have the following characteristics that are important to keep in mind in diagnostic scenarios.
+
+### Trimming as a first-class feature
+
+Unused code in a native AOT application is stripped out of the final binary. Any unbounded reflection used by the application, including its dependent libraries, is trimmed. The native AOT compiler will generate warnings in such cases, and it's critical that these warnings are fixed. For example, if you suppress any warnings without careful analysis, it can lead to hard to debug production failures.
```

Comment:
```suggestion
Unused code in a native AOT application is stripped out of the final binary. Any unbounded reflection used by the application, including its dependent libraries, is trimmed. The native AOT compiler will generate warnings in such cases, and it's critical that these warnings are fixed. For example, if you suppress any warnings without careful analysis, it can lead to hard-to-debug production failures.
```